app/src/main/java/com/gmail/nowak/wjw/popularmovies/data/repository/MoviesRepository.java

```diff
@@ -0,0 +1,324 @@
+package com.gmail.nowak.wjw.popularmovies.data.repository;
+
+import android.app.Application;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import com.gmail.nowak.wjw.popularmovies.AppExecutors;
+import com.gmail.nowak.wjw.popularmovies.BuildConfig;
+import com.gmail.nowak.wjw.popularmovies.data.db.AppDatabase;
+import com.gmail.nowak.wjw.popularmovies.data.model.api.ApiMovie;
+import com.gmail.nowak.wjw.popularmovies.data.model.api.ApiResponseMovieListObject;
+import com.gmail.nowak.wjw.popularmovies.data.model.api.ApiResponseReviewObject;
+import com.gmail.nowak.wjw.popularmovies.data.model.api.ApiResponseVideoObject;
+import com.gmail.nowak.wjw.popularmovies.data.model.api.ApiReview;
+import com.gmail.nowak.wjw.popularmovies.data.model.api.ApiVideo;
+import com.gmail.nowak.wjw.popularmovies.data.model.local.FavouriteMovie;
+import com.gmail.nowak.wjw.popularmovies.network.TheMovieDataBaseOrgAPI;
+import com.gmail.nowak.wjw.popularmovies.utils.NetworkUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import okhttp3.HttpUrl;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+import timber.log.Timber;
+
+import static com.gmail.nowak.wjw.popularmovies.utils.NetworkUtils.POPULARITY_TAG_TITLE;
+import static com.gmail.nowak.wjw.popularmovies.utils.NetworkUtils.TOP_RATED_TAG_TITLE;
+
+public class MoviesRepository {
+
+    Retrofit retrofit;
+    TheMovieDataBaseOrgAPI theMovieDatabaseOrgAPI;
+    private static final Object LOCK = new Object();
+    private static MoviesRepository sInstance;
+    private MutableLiveData<ApiResponseMovieListObject> topRatedMoviesResponseLD = new MutableLiveData<>();
+    private MutableLiveData<ApiResponseMovieListObject> popularMovieResponseLD = new MutableLiveData<>();
+    private LiveData<List<FavouriteMovie>> favMoviesData;// = new MutableLiveData<>();
+    private MutableLiveData<ApiMovie> apiMovieDetails = new MutableLiveData<>();
+    public AppDatabase database;
+    OkHttpClient okHttpClient;
+
+    public static MoviesRepository getInstance(Application application) {
+        if (sInstance == null) {
+            synchronized (LOCK) {
+                Timber.d("Creating new repository instance");
+                sInstance = new MoviesRepository(application);
+            }
+        }
+        Timber.d("Getting the repository instance");
+        return sInstance;
+    }
+
+    private MoviesRepository(Application application) {
```

Comment:
In constructor pass (TheMovieDataBaseOrgAPI theMovieDatabaseOrgAPI, AppDatabase database)
Create okHttpClient, retrofit and theMovieDatabaseOrgAPI somewhere else (most probably somewhere in onCreate in Application - but extract it to another file)